src/sha256.rs

```diff
@@ -129,6 +130,59 @@ impl HashTrait for Hash {
     }
 }
 
+/// Output of the SHA256 hash function
+#[derive(Copy, Clone, PartialEq, Eq, Default, PartialOrd, Ord, Hash)]
+pub struct Midstate([u8; 32]);
+
+hex_fmt_impl!(Debug, Midstate);
+hex_fmt_impl!(Display, Midstate);
+hex_fmt_impl!(LowerHex, Midstate);
+index_impl!(Midstate);
+serde_impl!(Midstate, 32);
+borrow_slice_impl!(Midstate);
+
+impl Midstate {
+    /// Length of the midstate, in bytes.
+    const LEN: usize = 32;
+
+    /// Flag indicating whether user-visible serializations of this hash
+    /// should be backward. For some reason Satoshi decided this should be
+    /// true for `Sha256dHash`, so here we are.
+    const DISPLAY_BACKWARD: bool = false;
+
+    /// Copies a byte slice into the [Midstate] object.
+    pub fn from_slice(sl: &[u8]) -> Result<Midstate, Error> {
+        if sl.len() != Self::LEN {
+            Err(Error::InvalidLength(Self::LEN, sl.len()))
+        } else {
+            let mut ret = [0; 32];
+            ret.copy_from_slice(sl);
+            Ok(Midstate(ret))
+        }
+    }
+
+    /// Unwraps the [Midatate] and returns the underlying byte array.
+    pub fn into_inner(self) -> [u8; 32] {
+        self.0
+    }
+
+    /// Produce a [Midstate] from the current state of a SHA246 engine.
+    pub fn from_sha256_engine(e: HashEngine) -> Midstate {
+        Midstate(e.midstate())
+    }
+}
+
+impl hex::FromHex for Midstate {
+    /// Parses a hex string as a hash object
+    fn from_hex(s: &str) -> Result<Self, Error> {
+        if s.len() != 2 * Self::LEN {
+            return Err(Error::InvalidLength(2 * Self::LEN, s.len()));
+        }
+
+        Ok(Self(hex::FromHex::from_hex(s)?))
```

Comment:
this is a compile error on 1.22

Reply:
I can compile on 1.22 locally after I updated.